br/pkg/lightning/mydump/region.go

```diff
@@ -302,6 +312,53 @@ func makeSourceFileRegion(
 	return []*TableRegion{tableRegion}, []float64{float64(fi.FileMeta.FileSize)}, nil
 }
 
+func SampleAndGetAvgRowSize(
+	fileInfo *FileInfo,
+	cfg *config.Config,
+	ioWorkers *worker.Pool,
+	store storage.ExternalStorage,
+) (int64, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	reader, err := store.Open(ctx, fileInfo.FileMeta.Path)
+	if err != nil {
+		return 0, nil
+	}
+	var parser Parser
+	switch fileInfo.FileMeta.Type {
```

Comment:
We should handle all possible file types here.

Reply:
Do we solely support 3 types of source files:
1. csv
2. sql
3. parquet

What else should I handle?

Reply:
It should be no more. It's better to return an error for unknown file types.

Reply:
Compared to csv, sql and parquet have some subtle differences. Please check whether we can apply the algorithm to them.

Reply:
parquet files won't be sampled in this way. They have provisioned an accurate number of lines.

FYI: https://github.com/pingcap/tidb/blob/0f85650aa43cb60a77ce507e02203db45584e914/br/pkg/lightning/mydump/region.go#L307-L335

I'll handle the unexpected file types then.